docs/source/format/Glossary.rst

```diff
@@ -0,0 +1,202 @@
+       corresponds to the :term:`parent type` and the child array(s)
+       correspond to the :term:`child type(s) <child type>`.  For
+       example, a ``List[Int32]``-type parent array has an
+       ``Int32``-type child array.
+
+   child type
+   parent type
+       In a :term:`nested type`, the nested type is the parent type,
+       and the child type(s) are its parameters.  For example, in
+       ``List[Int32]``, ``List`` is the parent type and ``Int32`` is
+       the child type.
+
+   chunked array
+       A *discontiguous*, *one-dimensional* sequence of values with
+       known length where all values have the same type.  Consists of
+       zero or more :term:`arrays <array>`, the "chunks".
+
+       Chunked arrays are discontiguous in the sense that iterating
+       the values of a chunked array may require iterating through
+       different buffers for different indices.
+
+       Not part of the columnar format; this term is specific to
+       certain language implementations of Arrow (primarily C++ and
+       its bindings).
+
+       .. seealso:: :term:`record batch`, :term:`table`
+
+   complex type
+   nested type
+       A :term:`data type` whose structure depends on one or more
+       other :term:`child data types <child type>`. For instance,
+       ``List`` is a nested type that has one child.
+
+       Two nested types are equal if and only if their child types are
+       also equal.
+
+   data type
+   type
+       A type that a value can take, such as ``Int8`` or
+       ``List[Utf8]``. The type of an array determines how its values
+       are laid out in memory according to :doc:`./Columnar`.
+
+       .. seealso:: :term:`nested type`, :term:`primitive type`
+
+   dictionary
+       An array of values that accompany a :term:`dictionary-encoded
+       <dictionary-encoding>` array.
+
+   dictionary-encoding
+       An array that stores its values as indices into a
+       :term:`dictionary` array instead of storing the values
+       directly.
+
+       .. seealso:: :ref:`dictionary-encoded-layout`
+
+   extension type
+   storage type
+       A user-defined :term:`data type` that adds additional semantics
+       to an existing data type.  This allows implementations that do
+       not support a particular extension type to still handle the
+       underlying data type (the "storage type").
+
+       For example, a UUID can be represented as a 16-byte fixed-size
+       binary type.
+
+       .. seealso:: :ref:`format_metadata_extension_types`
+
+   field
+       A column in a :term:`schema`.  Consists of a field name, a
+       :term:`data type`, a flag indicating whether the field is
+       nullable or not, and optional key-value metadata.
+
+   IPC format
+       A specification for how to serialize Arrow data, so it can be
+       sent between processes/machines, or persisted on disk.
+
+       .. seealso:: :term:`IPC file format`, :term:`IPC streaming
+                    format`
```

Comment:
I am not _fully_ sure this line break in inside the :term: will work for sphinx?

Reply:
I rendered it just now and it seems fine:
![image](https://user-images.githubusercontent.com/327919/171848278-89de8deb-33b1-453d-8ebd-a348332eacbc.png)
that said I pushed a line break anyways